vite.config.ts

```diff
@@ -1,23 +1,58 @@
+import process from "node:process";
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
+import type { ManifestOptions, VitePWAOptions } from "vite-plugin-pwa";
 import { VitePWA } from "vite-plugin-pwa";
+import replace from "@rollup/plugin-replace";
+
+const pwaOptions: Partial<VitePWAOptions> = {
+  mode: "development",
+  base: "/",
+  includeAssets: ["favicon.svg"],
+  manifest: {
+    name: "notes pwa",
+    short_name: "npwa",
+    theme_color: "#ffffff",
+  },
+  devOptions: {
+    enabled: process.env.SW_DEV === "true",
+    /* when using generateSW the PWA plugin will switch to classic */
+    type: "module",
+    navigateFallback: "index.html",
+  },
+};
+
+const replaceOptions = { __DATE__: new Date().toISOString() };
+const claims = process.env.CLAIMS === "true";
+const reload = process.env.RELOAD_SW === "true";
+const selfDestroying = process.env.SW_DESTROY === "true";
+
+if (process.env.SW === "true") {
+  pwaOptions.srcDir = "src";
+  pwaOptions.filename = claims ? "claims-sw.ts" : "prompt-sw.ts";
+  pwaOptions.strategies = "injectManifest";
+  (pwaOptions.manifest as Partial<ManifestOptions>).name =
+    "PWA Inject Manifest";
+  (pwaOptions.manifest as Partial<ManifestOptions>).short_name = "PWA Inject";
+}
+
+if (claims) pwaOptions.registerType = "autoUpdate";
+
+if (reload) {
+  // @ts-expect-error just ignore
+  replaceOptions.__RELOAD_SW__ = "true";
+}
+
+if (selfDestroying) pwaOptions.selfDestroying = selfDestroying;
 
-// https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [
-    VitePWA({
-      registerType: "autoUpdate",
-      manifest: {
-        name: "Notes PWA",
-        short_name: "NPWA",
-      },
-    }),
-    react(),
-  ],
+  // base: process.env.BASE_URL || 'https://github.com/',
   build: {
+    sourcemap: process.env.SOURCE_MAP === "true",
     outDir: "build",
```

Comment:
Probably want to add `manifest: true` here. It will ensure a manifest file is generated to support your app.
[Reference](https://vitejs.dev/guide/backend-integration.html)